internal/replay/replay_http_server.go

```diff
@@ -131,3 +147,109 @@ func (r *ReplayHTTPServer) writedResponse(w http.ResponseWriter, resp *store.Rec
 	_, err := w.Write(resp.Body)
 	return err
 }
+
+func (r *ReplayHTTPServer) proxyWebsocket(w http.ResponseWriter, req *http.Request, chunks []string) {
+	clientConn, err := r.upgradeConnectionToWebsocket(w, req)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error proxying websocket: %v", err), http.StatusInternalServerError)
+		return
+	}
+	defer clientConn.Close()
+	replayWebsocket(clientConn, chunks)
+}
+
+func (r *ReplayHTTPServer) loadWebsocketChunks(sha string) ([]string, error) {
+	responseFile := filepath.Join(r.recordingDir, sha+".websocket")
+	fmt.Printf("loading websocket response from : %s\n", responseFile)
+	bytes, err := os.ReadFile(responseFile)
+	var chunks = make([]string, 0)
+	if err != nil {
+		fmt.Printf("Error loading websocket response: %v\n", err)
+		return chunks, err
+	}
+
+	i := 0
+	response := string(bytes)
+	for i < len(response) {
+		// Extracts prefix
+		prefix := response[i]
+		if prefix != '>' && prefix != '<' {
+			return nil, fmt.Errorf("invalid message prefix at position %d: expected '>' or '<', got '%c'", i, prefix)
+		}
+		i++ // Move cursor past prefix.
+
+		// Extracts chunk length
+		numStart := i
+		for i < len(response) && unicode.IsDigit(rune(response[i])) {
+			i++
+		}
+		numEnd := i
+		if numStart == numEnd {
+			return nil, fmt.Errorf("missing chunk length after prefix at position %d", numStart-1)
+		}
+		numStr := response[numStart:numEnd]
+		num, err := strconv.Atoi(numStr)
+		if err != nil {
+			return nil, fmt.Errorf("invalid chunk length '%s': %w", numStr, err)
+		}
+
+		// Extracts chunk
+		chunkStart := numEnd
+		chunkEnd := chunkStart + num
+		if chunkEnd > len(response) {
+			return nil, fmt.Errorf("chunk length %d at position %d exceeds response bounds", chunkEnd, chunkStart)
+		}
+		chunk := response[chunkStart : chunkEnd-1] // Remove the \n appended at the end of the chunk
+		chunks = append(chunks, string(prefix)+chunk)
+		i = chunkEnd
+	}
+	return chunks, nil
+}
+
+func replayWebsocket(conn *websocket.Conn, chunks []string) {
+	for _, chunk := range chunks {
+		if strings.HasPrefix(chunk, ">") {
```

Comment:
Did we include the "<" / ">" in the chunk string we're building in loadwebsocketchunks?